### ISSUE TITLE
Method decorators

### ISSUE THREAD

**Issue by dtothefp** [loganfsmyth/babel-plugin-transform-decorators-legacy#10]
Thanks for this plugin. It sucks that a lot of React infrastructure started using decorators and now Babel isn't supporting it. I'm a little confused from the docs and the thread on Phabricator if decorating methods is supported by this plugin. 

In my app I have this decorator for toggling classes on the react render method. Works fine with Babel 5 but chokes with Babel 6. Please let me know if I'm doing something wrong or if this behavior is not supported

```
//.babelrc
{
  "presets": ["react", "es2015", "stage-0"],
  "env": {
    "development": {
      "plugins": [
        "rewire",
        "transform-runtime",
        "transform-decorators-legacy",
        "typecheck",
        ["react-transform",
          {
          "transforms": [{
            "transform": "react-transform-hmr",
            "imports": ["react"],
            "locals": ["module"]
          }, {
            "transform": "react-transform-catch-errors",
            "imports": ["react", "redbox-react"]
          }]
        }]
      ]
    },
    "production": {
      "plugins": [
        "transform-runtime",
        "transform-decorators-legacy",
        "typecheck"
      ]
    }
  }
}
```

``` js
//decorator.js
import React from 'react';
import _ from 'lodash';
import cx from 'classnames';

export default function(...args) {
  /*eslint no-use-before-define:0*/
  let decArgs;

  if (args.length === 1) {
    //if the decorator is called with an `omit` argument then return the function
    decArgs = args[0];
    return decorate;
  } else if (args.length === 3) {
    //if the decorator is called with by the method with no initial arguments
    return decorate.apply(null, args);
  }

  function decorate(target, key, descriptor) {
    return {
      ...descriptor,
      value(...args) {
        const {someFluxProp, anotherFluxProp} = this.props;
        const {omit} = decArgs || {};
        const Comp = descriptor.value.apply(this, args);

        const className = cx(_.assign({}, makeClasses(Comp.props.className), _.omit({
          'step-current': anotherFluxProp,
          'is-hidden': someFluxProp
        }, omit)));

        return React.cloneElement(Comp, {className});
      }
    };
  }
}
```

``` jsx
SomeClass extends React.Component {
    @addClassDec({omit: 'is-hidden'})
    render() {
       return <div className="whatevs" />
    }
}
```

When I do this I get output:

``` sh
ERROR in ./src/js/components/sample.jsx
Module parse failed: /Users/davidfox-powell/dev/frontend-boilerplate/node_modules/babel-loader/index.js?{"presets":["react","es2015","stage-0"],"plugins":["transform-decorators-legacy","transform-runtime","typecheck",["react-transform",{"transforms":[{"transform":"react-transform-hmr","imports":["react"],"locals":["module"]},{"transform":"react-transform-catch-errors","imports":["react","redbox-react"]}]}]]}!/Users/davidfox-powell/dev/frontend-boilerplate/node_modules/eslint-loader/index.js!/Users/davidfox-powell/dev/frontend-boilerplate/src/js/components/sample.jsx Line 3: Unexpected token
You may need an appropriate loader to handle this file type.
| 'use strict';
|
| import _Object$defineProperty from 'babel-runtime/core-js/object/define-property';
|
| var _dec, _dec2, _class, _desc, _value, _class2, _class3, _temp;
```

**Comment by loganfsmyth** [comment]
Could you make sure you are on version 1.3.1, not 1.3.0?

**Comment by dtothefp** [comment]
@loganfsmyth awesome, that seemed to work. It was working on `1.3.0` if I took `transform-runtime` out of presets but works now with runtime :smile:....thanks!

**Comment by loganfsmyth** [comment]
Yup, just a bug in the initial 1.3.x release.

**Comment by dtothefp** [comment]
@loganfsmyth I believe this issue related to the `runtime` has been re-introduced in `1.3.2` :cry: 

``` sh
WARNING in ./src/js/components/quick-card/index.jsx
Module parse failed: /Users/davidfox-powell/dev/donate-frontend/node_modules/@hfa/bundle-wrapper-loader/index.js?prefix=/Users/davidfox-powell/dev/donate-frontend/gulp/tasks/webpack/bundle-wrapper/banner.js&postfix=/Users/davidfox-powell/dev/donate-frontend/gulp/tasks/webpack/bundle-wrapper/footer.js&delimiter=
&remove=strict!/Users/davidfox-powell/dev/donate-frontend/node_modules/babel-loader/index.js?{"presets":["react","es2015","stage-0"],"plugins":["transform-runtime","transform-decorators-legacy","typecheck"]}!/Users/davidfox-powell/dev/donate-frontend/node_modules/eslint-loader/index.js!/Users/davidfox-powell/dev/donate-frontend/src/js/components/quick-card/index.jsx Line 14: Unexpected token
You may need an appropriate loader to handle this file type.
|
|
| import _Object$keys from 'babel-runtime/core-js/object/keys';
|
| var _dec, _dec2, _class, _desc, _value, _class2, _class3, _temp;
 @ ./src/js/components/sequential-form/index.jsx 107:17-41
```

**Comment by loganfsmyth** [comment]
Ugh that's frustrating. Published a 1.3.3 to fix it.

**Comment by dtothefp** [comment]
Cool thanks for the quick response
